src/validation.cpp

```diff
@@ -1294,15 +1295,11 @@ void CoinsViews::InitCache()
 CChainState::CChainState(CTxMemPool* mempool,
                          BlockManager& blockman,
                          CEvoDB& evoDb,
-                         const std::unique_ptr<llmq::CChainLocksHandler>& clhandler,
-                         const std::unique_ptr<llmq::CInstantSendManager>& isman,
-                         const std::unique_ptr<llmq::CQuorumBlockProcessor>& quorum_block_processor,
+                         const std::unique_ptr<LLMQContext>& llmq_ctx,
                          std::optional<uint256> from_snapshot_blockhash)
     : m_mempool(mempool),
       m_params(::Params()),
-      m_clhandler(clhandler),
-      m_isman(isman),
-      m_quorum_block_processor(quorum_block_processor),
+      m_llmq_ctx(llmq_ctx),
```

Comment:
I'm personally leaning towards the method of initialization you've done with `BlockAssembler`

```
      quorum_block_processor(*llmq_ctx.quorum_block_processor),
      m_clhandler(*llmq_ctx.clhandler),
      m_isman(*llmq_ctx.isman),
```

The purpose of this PR is to reduce the exceedingly long argument lists but I still feel they should resolved as distinct members within the object. If another `LLMQContext` member has to be accessed, the developer should be made to add a new member, making it glaringly clear that the object now depends on another subsystem instead of being able to subtly add dependencies by directly accessing `LLMQContext`

Having three more arguments are unwieldily but being able to access the rest of `LLMQContext`'s members, I think, is even more so.

Reply:
`BlockAssember` should not assume that `quorum_block_processor`, `m_clhandler` can be created independantly and used independantly. If there's any cases when it should be indeed different there are already several difficulties:
 - some of these members are global variables and still not de-globalised. So, at current moment, there should not be created and used independantly and of this objects ever.
 - even if it would be deglobalized - create a new LLMQContext and pass *for sure consistent* data rather bunch of variables that has hidden connections between.

Everything that is outside of `llmq` sub-module should operate only with public interfaces of `llmq` but not with particular implementation how it is made now. It's out-of-scope of current PR and I aim to do that on future de-globalization PRs.

Reply:
I think I agree with kittywiskers here that it's better to keep these multiple members to make it more clear what is actually being used and where the dependencies are. @UdjinM6 ?